### PR TITLE
Creating register and login endpoints

### PR DESCRIPTION
**Pull request by swathivottigunta**
creating the register(/student/register) and login(student/login) endpoints. Please do provide your view points on schema creation, register and login endpoints.

### CODE REVIEW

File: server/middleware/validator.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      }
+      return true;
+    })
+  ];
+};
+
+exports.loginValidationRules = () => {
+  return [
+    body("email").isEmail().withMessage("Please include a valid email"),
+    body("password").not().isEmpty().withMessage("Name is required")
```
Comment (ugur-akin) [inline server/middleware/validator.js]: "Name" should probably be "Password"?

**Review by sina-jamshidi — APPROVED** [review]
This looks great!

**Comment by nicktran6997** [comment]
Hey it looks good! btw I'm probably gonna rearrange the file directories of the user routes into api/ just for consistency with my code after I merge it

File: server/routes/user.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+// @route  POST user/register
+// @desc   register user
+// @access public
+
+router.post(
+  "/register",
+  registerValidationRules(),
+  validate,
+  async (req, res) => {
```
Comment (nicktran6997) [inline server/routes/user.js:26]: quick question about the middleware, does the response of each middleware function gets passed into the next() right? Sorry am relatively newish to express, Also, nice async usage, I've been trying to incorporate that in my routes for code consistency,

**Review by nicktran6997 — APPROVED** [review]
Looks really solid overall

File: server/routes/user.js (hunk 2 above)
Comment (swathivottigunta) [reply server/routes/user.js:26]: 1. registerValidationRules() -> this function returns the array (validation on each field)
2. validate -> this is the middleware function that validates the fields based on the rules, explicitly need to specify next() to continue

**Pull request merged**